test/graph_test.cpp

```diff
@@ -88,3 +88,49 @@ TEST(GraphTestSuite, TestCorrectnessOnSmallSparseGraphs) {
     g.connected_components();
   }
 }
+
+TEST(GraphTestSuite, TestCorrectnessOfReheating) {
+  int num_trials = 1;
+  while (num_trials--) {
+    generate_stream({1024,0.002,0.5,0,"./sample.txt","./cum_sample.txt"});
+    ifstream in{"./sample.txt"};
+    Node n, m;
+    in >> n >> m;
+    Graph g{n};
+    int type, a, b;
+    while (m--) {
+      in >> type >> a >> b;
+      if (type == INSERT) {
+        g.update({{a, b}, INSERT});
+      } else g.update({{a, b}, DELETE});
+    }
+    g.set_cum_in("./cum_sample.txt");
+    g.write_binary("./out_temp.txt");
+    auto g_res = g.connected_components();
+    printf("number of CC = %u\n", g_res.size());
+
+    Graph reheated {"./out_temp.txt"};
+
+    // for (int i = 0; i < log2(n); i++) {
+    //   ASSERT_EQ(g.supernodes[i]->logn, reheated.supernodes[i]->logn);
+    //   //ASSERT_EQ(g.supernodes[i]->idx, reheated.supernodes[i]->idx);
+    //   for (int j = g.supernodes[i]->idx;j < g.supernodes[i]->logn;++j) {
+    //     if(!(*g.supernodes[i]->get_sketch(j) == *reheated.supernodes[i]->get_sketch(j))) {
+    //       printf("sketches at index %i do not match of supernode %i\n", j, i);
+    //       ASSERT_EQ(1, 0);
+    //     }
+    //   }
+    // }
```

Comment:
why is this commented out?

Reply:
Because it was a part of our debugging not a part of the actual test. I will remove the code.